imports/server/WorkerPool.ts

```diff
@@ -0,0 +1,169 @@
+/* eslint-disable camelcase */
+import child_process from 'child_process';
+import { WebApp } from 'meteor/webapp';
+import portscanner from 'portscanner';
+
+type HandledSignal = 'SIGINT' | 'SIGHUP' | 'SIGTERM';
+
+export type Worker = {
+  process: child_process.ChildProcess;
+  id: number;
+  port: number;
+}
+
+// worker processes should notify their parent once they're ready to accept
+// requests
+if (process.env.CLUSTER_WORKER_ID) {
+  WebApp.onListening(() => {
+    if (process.send) {
+      process.send({
+        type: 'ready',
+      });
+    }
+  });
+}
+
+export default class WorkerPool {
+  exec: string;
+
+  args: string[];
+
+  workers: Worker[];
+
+  workersMap: Record<number, Worker>;
+
+  ids: number;
+
+  closed: boolean;
+
+  recentReconnects: number;
+
+  lastReconnectAt: number;
+
+  static create(size: number): WorkerPool | undefined {
+    if (process.env.CLUSTER_WORKER_ID) {
+      return undefined;
+    }
+    return new WorkerPool(size);
+  }
+
+  constructor(size: number) {
+    this.exec = process.argv[1];
+    this.args = process.argv.slice(2);
+    this.workers = [];
+    this.workersMap = {};
+    this.ids = 0;
+    this.closed = false;
+    for (let i = 0; i < size; i++) {
+      this.createWorker();
+    }
+    this.recentReconnects = 0;
+    this.lastReconnectAt = 0;
+
+    (['SIGINT', 'SIGHUP', 'SIGTERM'] as const).forEach((signal: HandledSignal) => {
+      process.once(signal, this.cleanup.bind(this));
+    });
```

Comment:
Hmm. Is this problematic given what we ran into with #483?

Reply:
I don't think so?  This is logic that previously existed in the dependency, and the `cleanup` method below also re-signals itself after the hook.